### PR TITLE
fix-drop-advisory

### PR DESCRIPTION
**Pull request by Ximinhan**
move advisory to NEW_FILES before drop builds
test build https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/ximhan/job/build%252Fdrop_advisories/4/console

### CODE REVIEW

File: elliottlib/cli/advisory_drop_cli.py [hunk 1]
```diff
@@ -24,13 +24,17 @@ def advisory_drop_cli(advisory):
         click.echo(f'Advisory {advisory} is already dropped')
         return
 
+    # move advisory status to NEW_FILES
+    if adv.errata_state != "NEW_FILES":
+        adv.setState("NEW_FILES")
+        adv.commit()
+
     # Remove bugs and builds before dropping the advisory
+    adv = Erratum(errata_id=advisory)
```
Comment (thegreyd) [inline elliottlib/cli/advisory_drop_cli.py:33]: I don't think we need this, commit() will probably do an internal refresh. can you test without this?

File: elliottlib/cli/advisory_drop_cli.py [hunk 2]
```diff
@@ -44,7 +48,8 @@ def advisory_drop_cli(advisory):
     data = 'utf8=%E2%9C%93&reason=Dropping+unused+advisory%21&commit=Dropping+unused+advisory'
     headers = {"Content-Type": "text/plain"}
     r = requests.post(url, auth=HTTPSPNEGOAuth(), data=data, headers=headers)
-    if not r.ok:
+    adv = Erratum(errata_id=advisory)
+    if adv.errata_state != "DROPPED_NO_SHIP":
```
Comment (thegreyd) [inline elliottlib/cli/advisory_drop_cli.py:52]: I don't like refetching here, we should trust the return code. I would like to print the return error output if we get any

Comment (Ximinhan) [reply elliottlib/cli/advisory_drop_cli.py:52]: the return code is not reliable, see https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/ximhan/job/build%252Fdrop_advisories/3/console it returns 500 but actually advisory already dropped, we don't need to check the return code as long as the advisory status is dropped

**Review by joepvd — APPROVED** [review]
/lgtm

**Pull request merged**